### PR TITLE
init ci setup

### PR DESCRIPTION
**Pull request by AlexJoz**

### CODE REVIEW

File: .github/workflows/initi.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+           context: .
+           file: ./Dockerfile
+#          push: ${{ github.event_name != 'pull_request' }}
+           push: true
+           tags: ${{ steps.prep.outputs.tag }}
+           cache-from: type=local,src=/tmp/.buildx-cache
+           cache-to: mode=max,type=local,dest=/tmp/.buildx-cache
+
+
+  test_1:
```
Comment (ibayer) [inline .github/workflows/initi.yaml:47]: This `test_i` steps map directly to the official nbdev test. Can we use the same task/job names
as in gitlab?

* read_notebooks:
* check_notebooks_clean:
* check_diff_lib_nbs:
* un_tests:

File: .github/workflows/initi.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.REP_PAT }}
+
+      - name: Pull image from DockerHub
+        run: docker pull ${{ steps.prep.outputs.tag }}
+
+      - name: Run image in container
```
Comment (ibayer) [inline .github/workflows/initi.yaml:92]: When going through the CI output steps it wasn't clear to me that `Run image in container` step contains the main information.
I would suggest to to use the job name also for the task name (or something very similar).
`Pull image from DockerHub` -> `"Check Notebooks Clean"` The same applies to the other jobs.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # ipyannotator - the infinitely hackable annotation framework
 
-
+![CI-Badge](https://github.com/palaimon/ipyannotator/workflows/CI/badge.svg)
```
Comment (ibayer) [inline README.md:3]: image

Badge doesn't seem to render properly on your branch. Will this change as soon as we merge? (path seems to point to palaimon repo)

**Review by ibayer — CHANGES_REQUESTED** [review]
@AlexJoz 
Very nice job. I just have some minor change requests.

File: .github/workflows/initi.yaml (hunk 1 above)
Comment (AlexJoz) [reply .github/workflows/initi.yaml]: :boom:

File: .github/workflows/initi.yaml (hunk 2 above)
Comment (AlexJoz) [reply .github/workflows/initi.yaml]: :boom:

File: README.md (hunk 3 above)
Comment (AlexJoz) [reply README.md:3]: I think it will render after you setup org repo and ci workflow run happen

**Pull request merged**